mewarpx/mewarpx/setups_store/diode_setup.py

```diff
@@ -317,35 +262,72 @@ def init_base(self):
             # immediately sets xmmin and ymmin to 0 when
             # warp.w3d.initdecompositionw3d() is called, so we work with that
             # offset here, though nothing should depend on it.
-            warp.w3d.xmmin = 0
-            warp.w3d.xmmax = 1.0
+            xmmin = 0
+            xmmax = 1.0
         else:
-            warp.w3d.xmmin = -self.PERIOD/2.0
-            warp.w3d.xmmax = self.PERIOD/2.0
+            xmmin = -self.PERIOD/2.0
+            xmmax = self.PERIOD/2.0
 
         if self.dim < 3 and not self.rz:
-            warp.w3d.ymmin = 0
-            warp.w3d.ymmax = 1.0
+            ymmin = 0
+            ymmax = 1.0
         else:
-            warp.w3d.ymmin = -self.PERIOD/2.0
-            warp.w3d.ymmax = self.PERIOD/2.0
+            ymmin = -self.PERIOD/2.0
+            ymmax = self.PERIOD/2.0
 
-        warp.w3d.zmmin = -self.OFFSET
-        warp.w3d.zmmax = self.D_CA + self.OFFSET
+        zmmin = -self.OFFSET
+        zmmax = self.D_CA + self.OFFSET
 
         # Grid parameters - set grid counts
         if self.dim == 1:
-            warp.w3d.nx = 0
+            nx = 0
         else:
-            warp.w3d.nx = int(round(self.PERIOD/self.RES_LENGTH))
+            nx = int(round(self.PERIOD/self.RES_LENGTH))
 
         if self.dim < 3 and not self.rz:
-            warp.w3d.ny = 0
+            ny = 0
         else:
-            warp.w3d.ny = int(round(self.PERIOD/self.RES_LENGTH))
+            ny = int(round(self.PERIOD/self.RES_LENGTH))
 
-        warp.w3d.nz = int(round((warp.w3d.zmmax - warp.w3d.zmmin)
-                                / self.RES_LENGTH))
+        nz = int(round((zmmax - zmmin)/self.RES_LENGTH))
+
+        # create the grid
+        if self.dim == 1:
+            raise NotImplementedError("1D grid is not yet implemented in mewarpx")
+        elif self.dim == 2:
+            self.grid = picmi.Cartesian2DGrid(
+                            number_of_cells=[nx, nz],
+                            lower_bound=[xmmin, zmmin],
+                            upper_bound=[xmmax, zmmax],
+                            bc_xmin='periodic',
+                            bc_xmax='periodic',
+                            bc_zmin='dirichlet',
+                            bc_zmax='dirichlet',
+                            warpx_potential_hi_x=self.V_ANODE,
+                            warpx_potential_low_x=self.V_CATHODE,
+                            lower_boundary_conditions_particles=['periodic', 'absorbing'],
+                            upper_boundary_conditions_particles=['periodic', 'absorbing'],
+                            moving_window_velocity=None,
+                            warpx_max_grid_size=nx//4
```

Comment:
For now probably just make `MAX_GRID_SIZE` a new class variable and assign it here? or call it `MAX_GRID_SIZE_FACTOR` and divide nz by that (nz, not nx, I'm guessint is what you want)